### PR TITLE
Issue #7610: Update doc for HideUtilityClassConstructor

### LINKED ISSUE #7610: Update doc for HideUtilityClassConstructor
https://checkstyle.org/config_design.html#HideUtilityClassConstructor

Add example for each existing config to show what code is ok and what code cause violation.
Good example is https://checkstyle.org/config_naming.html#AbstractClassName
Look at referenced commits of #5832, to see what and how should be changed in code.

Please play with module/check by means of https://checkstyle.org/cmdline.html to understand how it works.

------------

PLEASE READ https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

### PR DESCRIPTION
**Pull request by sulkykookie**
Fixes #7610
PR6

Default Example:
```
$ cat config3.xml
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="HideUtilityClassConstructor">
    </module>
  </module>
</module>

$ cat Test3.java
class Test { // violation, class only has a static method and a constructor

  public Test() {
  }

  public static void fun() {
  }
}

class Foo { // OK

  private Foo() {
  }

  static int n;
}

class Bar { // OK

  protected Bar() {
    // prevents calls from subclass
    throw new UnsupportedOperationException();
  }
}

class UtilityClass { // violation, class only has a static field

  static float f;
}

$ java -jar checkstyle-8.30-all.jar -c config3.xml Test3.java
Starting audit...
[ERROR] C:\Users\Shrey\Desktop\Checkstyle jar\Test3.java:1:1: Utility classes should not have a public or default constructor. [HideUtilityClassConstructor]
[ERROR] C:\Users\Shrey\Desktop\Checkstyle jar\Test3.java:26:1: Utility classes should not have a public or default constructor. [HideUtilityClassConstructor]
Audit done.
Checkstyle ends with 2 errors.
```

The check provides no properties.

### CODE REVIEW

**Comment by pbludov** [comment]
@romani I can't understand why in this check `package` scope is prohibited, but less strict `protected` scope is allowed?
This behaviour is from the initial [commit](https://github.com/checkstyle/checkstyle/blob/ae9edbd4a8645c8cc1a99ad6890dd0006700d1cb/src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java#L231)

**Comment by sulkykookie** [comment]
@strkkk Please review

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java [hunk 1]
```diff
@@ -57,6 +57,43 @@
  * <pre>
  * &lt;module name=&quot;HideUtilityClassConstructor&quot;/&gt;
  * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class Test { // violation, class only has a static method
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java:64]: `has a static method` and constructor

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java [hunk 2]
```diff
@@ -57,6 +57,43 @@
...
+ *
+ * class Bar { // OK
+ *
+ *   protected Bar() {
+ *     // prevents calls from subclass
+ *     throw new UnsupportedOperationException();
+ *   }
+ * }
+ *
+ * class testClass { // violation, class only has a static field
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java:89]: Please change name, class name should start from uppercase

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java [hunk 3]
```diff
@@ -57,6 +57,43 @@
...
+ *
+ *   public testClass() {
+ *   }
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java:92]: remove constructor to make example without constructor

**Comment by strkkk** [comment]
ok to merge after ci is ok

**Comment by sulkykookie** [comment]
Why am I getting wercker error in this and #7922?

**Pull request merged**